### PR TITLE
Document deployment to Seldon Core and KServe

### PR DESCRIPTION
**Pull request by adriangonz**
## Changelog
- Add Deployment section, including instructions to deploy MLServer models with both Seldon Core and KServe

### CODE REVIEW

File: docs/user-guide/deployment/kserve.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+| XGBoost      | [MLServer XGBoost](../../runtimes/xgboost) | `xgboost`              | [XGBoost Serving Runtime](https://kserve.github.io/website/modelserving/v1beta1/xgboost/)    |
+
+Note that, on top of the ones shown above (backed by MLServer), KServe also
+provides a [wider
+set](https://kserve.github.io/website/modelserving/v1beta1/serving_runtime/) of
+serving runtimes.
+To see the full list, please visit the [KServe
+documentation](https://kserve.github.io/website/modelserving/v1beta1/serving_runtime/).
+
+## Custom Images
```
Comment (sakoush) [inline docs/user-guide/deployment/kserve.md:89]: add docker / container images?

File: docs/user-guide/deployment/kserve.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+These images then become self-contained model servers with your custom runtime.
+Therefore, it's easy to deploy them into your serving infrastructure leveraging
+KServe support for [custom
+runtimes](https://kserve.github.io/website/modelserving/v1beta1/custom/custom_model/#deploy-the-custom-predictor-on-kserve).
+
+### Usage
+
+The `InferenceService` manifest gives you full control over the containers used
+to deploy your machine learning model.
+This can be leveraged to poing your deployment to the [custom MLServer image
```
Comment (sakoush) [inline docs/user-guide/deployment/kserve.md:104]: `poing` spelling mistake?

File: docs/user-guide/deployment/kserve.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+### Supported Serving Runtimes
+
+As mentioned above, KServe offers support for built-in serving runtimes, some
+of which leverage MLServer as the inference server.
+Below you can find a table listing these runtimes, and the MLServer inference
+runtime that they correspond to.
+
+| Framework    | MLServer Runtime                           | KServe Serving Runtime | Documentation                                                                                |
+| ------------ | ------------------------------------------ | ---------------------- | -------------------------------------------------------------------------------------------- |
+| Scikit-Learn | [MLServer SKLearn](../../runtimes/sklearn) | `sklearn`              | [SKLearn Serving Runtime](https://kserve.github.io/website/modelserving/v1beta1/sklearn/v2/) |
```
Comment (sakoush) [inline docs/user-guide/deployment/kserve.md:79]: kserve doesnt support mlflow?

**Review by sakoush — APPROVED** [review]
LGTM

should we point to any example/notebook from seldon-core that illustrates a deployment as well?

File: docs/user-guide/deployment/kserve.md (hunk 1 above)
Comment (adriangonz) [reply docs/user-guide/deployment/kserve.md:89]: How do you mean?

File: docs/user-guide/deployment/kserve.md (hunk 3 above)
Comment (adriangonz) [reply docs/user-guide/deployment/kserve.md:79]: There isn't a top-level serving runtime exposing the MLServer one yet.

File: docs/user-guide/deployment/kserve.md (hunk 1 above)
Comment (sakoush) [reply docs/user-guide/deployment/kserve.md:89]: it was very minor comment about whether the title should be `Custom Docker Images`.

Comment (adriangonz) [reply docs/user-guide/deployment/kserve.md]: Oh I see! Good point, I've now renamed it to `Custom Runtimes`.

**Pull request merged**